src/main/java/frc/robot/subsystems/SparkMax.java

```diff
@@ -0,0 +1,44 @@
+package frc.robot.subsystems;
+
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+
+import edu.wpi.first.wpilibj.motorcontrol.MotorController;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import edu.wpi.first.wpilibj2.command.button.CommandXboxController;
+import frc.robot.Constants.ControlConstants;
+import frc.robot.Constants.OperatorConstants;
+import frc.robot.subsystems.Drivetrain;
+import frc.robot.RobotContainer;
+
+import edu.wpi.first.wpilibj.TimedRobot;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+
+import com.revrobotics.RelativeEncoder;
+import com.revrobotics.SparkMaxPIDController;
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.SparkMaxRelativeEncoder;
+import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+
+
+public class SparkMax extends SubsystemBase {
+    private static final int I = 0;
+    public CANSparkMax m;
+    public RelativeEncoder e;
+
+    public SparkMax() {
+        m = new CANSparkMax(0,MotorType.kBrushless);
+        e = m.getEncoder();
+ 
+    }
+@Override
+	public void periodic() { 
+        double v = e.getVelocity();
+        double p = e.getPosition();
+        double CPR = e.getCountsPerRevolution();           
+        SmartDashboard.putNumber("Velocity", v);
+        SmartDashboard.putNumber("Position", p);
+        SmartDashboard.putNumber("CountsPerRevolution", CPR);
+
+    }
```

Comment:
This is all great. You can also move the initialization of `e` and add all the velocity and position checks into `Drivetrain`. This is because you've already initialized the 4 motors there, and you can get the encoders from those motors. You should also look into adding a few methods that returns the velocity and/or position to a command outside of the function (think autonomous)